app/src/main/java/com/artemzin/qualitymatters/other/GsonTypeAdapterFactory.java

```diff
@@ -0,0 +1,18 @@
+package com.artemzin.qualitymatters.other;
+
+import com.artemzin.qualitymatters.api.entities.Item;
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+
+public class GsonTypeAdapterFactory implements TypeAdapterFactory {
```

Comment:
You are right. I will change to the factory generated by autovalue-gson. For some reason I have overlooked it in documentation.